lib/evmone/instructions_calls.cpp

```diff
@@ -20,6 +20,13 @@ evmc_status_code call(ExecutionState& state) noexcept
 
     state.stack.push(0);  // Assume failure.
 
+    if (state.rev >= EVMC_BERLIN && state.host.access_account(dst) == EVMC_COLD_ACCESS)
```

Comment:
Minimal improvement here would be to merge `access_account()` and `account_exists()`. I believe the account can have 3 statuses then: hot, cold, non-existing.

BTW, can a non-existing account become hot?

Reply:
Not sure that merging `access_account` with `account_exists` is worth it. It might save a function call, but the two are quite different: `account_exists` ([EIP-161](https://eips.ethereum.org/EIPS/eip-161)) is a constant operation that doesn't change anything, while `access_account` modifies `accessed_addresses` substate ([EIP-2929](https://eips.ethereum.org/EIPS/eip-2929)).

Reply:
`SELFDESTRUCT` puts non-existing beneficiary into a warm set, the same I thihk with calls, so warm/cold and existing/non-existing seem to be orthogonal.

One could also think that `account_exists` puts non-existing account into a warm set, it's used only in `SELFDESTRUCT` and calls which do it, and it would reflect reality of updating non-existence caches in clients.